cluster-autoscaler/cloudprovider/gke/gke_manager.go

```diff
@@ -467,6 +508,10 @@ func (m *gkeManagerImpl) forceRefresh() error {
 			glog.Errorf("Failed to fetch node pools: %v", err)
 			return err
 		}
+		if err := m.refreshLocations(); err != nil {
```

Comment:
Unless I'm missing something with this implementation we do 3 identical API calls in a row when calling forceRefresh() (a separate GET cluster call for nodepools, locations and limits). Can we somehow avoid that? Maybe cache cluster in GKEService or something? I'm fine accepting this and improving in follow up PR if you prefer it that way.

Reply:
Agreed, repetitive API calls are definitely something we can optimize. I'd prefer to do it in a separate PR since it's unrelated/bigger change. In the meantime, moved this call to NAP case.